systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java

```diff
@@ -69,6 +71,86 @@ void testProducerConsumer() {
         ClientUtils.waitForClientSuccess(OAUTH_CONSUMER_NAME, NAMESPACE, MESSAGE_COUNT);
     }
 
+    @Test
+    void testProducerConsumerAudienceTokenChecks() {
+        LOGGER.info("Update Kafka broker configuration to use OAuth2 access token audience checks");
+        KafkaResource.replaceKafkaResource(oauthClusterName, kafka -> {
+            kafka.getSpec().getKafka().setListeners(new ArrayOrObjectKafkaListenersBuilder()
+                    .addNewGenericKafkaListener()
+                    .withName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                    .withPort(9092)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(false)
+                    .withNewKafkaListenerAuthenticationOAuth()
+                    .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                    .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                    .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                    .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                    .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                    .withEnablePlain(true)
+                    .withCheckAudience(true)
+                    .withClientId("kafka-component")
+                    .withTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                    .endKafkaListenerAuthenticationOAuth()
+                    .endGenericKafkaListener().build());
+        });
```

Comment:
When you change configuration of Kafka, shouldn't you change it back to not affect other tests?

Reply:
Thanks! I forgot we don't deploy new kafka between tests :)